### ISSUE TITLE
After using multi-level nested routing, the component will render multiple times before it is successfully cached

### ISSUE THREAD

**Issue by ZHT131** [vuejs/vue-router-next#1102]
### Version
4.0.11

### Reproduction link
[github.com](https://github.com/ZHT131/brisk-admin/tree/dev)

### Steps to reproduce
router config
```
      {
        path: "/nested",
        component: "layout/index.vue",
        name: "nested",
        redirect: null,
        meta: { title: "nested", icon: "el-icon-menu" },
        alwaysShow: true,
        children: [
          {
            path: "menu",
            component: "layout/nestedComponent.vue",
            name: "menu",
            meta: { title: "menu", icon: "el-icon-menu" },
            redirect: null,
            alwaysShow: true,
            children: [
              {
                path: "menu2",
                component: "layout/nestedComponent.vue",
                name: "menu2",
                meta: { title: "menu2", icon: "el-icon-menu" },
                redirect: null,
                alwaysShow: true,
                children: [
                  {
                    path: "menu4",
                    component: "menu4/index.vue",
                    name: "menu4",
                    meta: { title: "menu4", icon: "el-icon-menu" },
                    redirect: null,
                    alwaysShow: false,
                  },
                ],
              },
              {
                path: "menu3",
                component: "menu3/index.vue",
                name: "menu3",
                meta: { title: "menu3", icon: "el-icon-menu" },
                redirect: null,
                alwaysShow: false,
              },
            ],
          },
          {
            path: "menu1",
            component: "menu1/index.vue",
            name: "menu1",
            meta: { title: "menu1", icon: "el-icon-menu" },
            redirect: null,
            alwaysShow: false,
          },
        ],
      },
```
layout/nestedComponent.vue

```
<template>
  <div>
    <router-view v-slot="{ Component }">
      <transition appear name="fade-transform" mode="out-in">
        <keep-alive>
          <component :is="Component" :key="key" />
        </keep-alive>
      </transition>
    </router-view>
  </div>
</template>

<script>
export default {
  data() {
    return {};
  },
  computed: {
    key() {
      return this.$route.path;
    },
  },
  methods: {},
};
</script>

<style>
</style>
```
![image](https://user-images.githubusercontent.com/22323594/132114656-c8dd1392-0174-4ff6-a063-73cf3b4727fe.png)

This will cause multiple levels of nested rendering of components. I currently render and execute the page life cycle twice in the fourth level, and the page can be cached successfully after switching back and forth several times.

### What is expected?
Load nested components only once and cache

### What is actually happening?
Load nested components multiple times and switch back and forth to cache

**Comment by posva** [comment]
This is too much for a repro. Can you try reproducing it for with only Vue, Vue router and vite? Open a new issue once you have it
